openmls/src/group/core_group/process.rs

```diff
@@ -34,14 +34,25 @@ impl CoreGroup {
         //  - ValSem003
         self.validate_framing(&message)?;
 
+        let epoch = message.epoch();
+
         // Checks the following semantic validation:
         //  - ValSem006
-        let decrypted_message = match message.wire_format() {
-            WireFormat::MlsPlaintext => DecryptedMessage::from_inbound_plaintext(message)?,
-            WireFormat::MlsCiphertext => {
+        //  - ValSem007 MembershipTag presence
+        let decrypted_message = match message.mls_message.body {
+            MlsMessageBody::Plaintext(plaintext) => {
+                // If the message is older than the current epoch, we need to fetch the correct secret tree first.
+                let message_secrets =
+                    self.message_secrets_for_epoch(epoch).map_err(|e| match e {
+                        SecretTreeError::TooDistantInThePast => ValidationError::NoPastEpochData,
+                        _ => LibraryError::custom("Unexpected return value").into(),
```

Comment:
Maybe have a message with a little more information about what went wrong? Something like "Unexpected error while retrieving message secrets for epoch".